homeassistant/components/philips_air_purifier/fan.py

```diff
@@ -0,0 +1,166 @@
+
+from .client import FanSpeed, Mode, ReliableClient, Status
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a configured Air Purifier by creating the Fan entity."""
+    client = hass.data[DOMAIN][config_entry.entry_id]
+
+    initial_data = {
+        "unique_id": config_entry.unique_id,
+        "name": config_entry.title,
+        "model": config_entry.data["model"],
+    }
+
+    entity = PurifierEntity(client, initial_data)
+    async_add_entities([entity])
+
+
+class PurifierEntity(FanEntity):
+    """Representation of a Philips air purifier."""
+
+    should_poll: bool = False
+
+    supported_features: int = SUPPORT_PRESET_MODE | SUPPORT_SET_SPEED
+    preset_modes: list[str] = [m.name for m in list(Mode) if m != Mode.Manual]
+    speeds = [FanSpeed.Speed1, FanSpeed.Speed2, FanSpeed.Speed3, FanSpeed.Turbo]
+    speed_count: int = len(speeds)
+
+    def __init__(self, client: ReliableClient, initial_data: dict[str, Any]) -> None:
+        """Initialize a PurifierEntity."""
+        self._client = client
+        self._status: Status | None = None
+        # If the device is turned off or not reachable for other reasons, we want the
+        # integration to be able to start up anyway and just be unavailable.
+        self._initial_data = initial_data
+
+    async def async_added_to_hass(self) -> None:
+        """Subscribe to device status updates."""
+        self._client.observe_status(id(self), self._set_status)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Unsubscribe from device status updates."""
+        self._client.stop_observing_status(id(self))
+
+    def _set_status(self, status: Status | None) -> None:
+        self._status = status
+        self.schedule_update_ha_state()
+
+    @property
+    def unique_id(self):
+        """Return the unique ID for this purifier."""
+        if self._status is None:
+            return self._initial_data["unique_id"]
+        return self._status.device_id
+
+    @property
+    def name(self):
+        """Return the purifier's name."""
+        if self._status is None:
+            return self._initial_data["name"]
+        return self._status.name
+
+    @property
+    def available(self):
+        """Return whether the purifier is available."""
+        return self._status is not None
+
+    @property
+    def icon(self) -> str | None:
+        """Use an air purifier icon instead of the default fan."""
+        return "mdi:air-purifier"
+
+    @property
+    def device_info(self):
+        """
+        Return device information for the purifier.
+
+        Given that the purifier is sometimes not responsive for 30 seconds ore more,
+        we can't query it during startup and can only use data from the config entry.
+        """
+        info = {
```

Comment:
Just set `self._attr_device_info` in your constructor. And use the DeviceInfo tuple `from homeassistant.helpers.entity import DeviceInfo`.